app/models/user.rb

```diff
@@ -106,6 +106,50 @@ def gpo_verification_pending_profile?
     gpo_verification_pending_profile.present?
   end
 
+  def suspended?
+    suspended_at.present?
+  end
+
+  def reinstated?
+    reinstated_at.present?
+  end
+
+  def suspend
```

Comment:
I would call these `suspend!` and `reinstate!` to make it clear they have side effects